java/events_slack_notifications/src/main/java/com/my_asana/app/MilestoneCelebratorySlackMessage.java

```diff
@@ -0,0 +1,141 @@
+package com.my_asana.app;
+
+import com.asana.Client;
+import com.asana.models.Event;
+import com.asana.models.Project;
+import com.asana.models.Workspace;
+import com.asana.models.Story;
+import com.asana.models.ResultBodyCollection;
+import com.asana.models.Task;
+import com.asana.errors.InvalidTokenError;
+
+import com.hubspot.slack.client.SlackClient;
+import com.hubspot.slack.client.SlackClientFactory;
+import com.hubspot.slack.client.SlackClientRuntimeConfig;
+import com.hubspot.slack.client.methods.params.chat.ChatPostMessageParams;
+import com.hubspot.slack.client.models.response.SlackError;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.FileInputStream;
+import java.util.List;
+
+public class MilestoneCelebratorySlackMessage {
+
+    private static String CHANNEL_ID = "#asana-notifications";
+
+    public static void main(String[] args) throws Exception {
+
+        Client client = getAsanaClient();
+        // SlackClient slackClient = getSlackClient();
+
+        String projectId = args[0];
+
+        // Attempt to get latest sync token.
+        File syncFile = new File("asana_sync_token_for_project_" + projectId);
+
+        // If the file does not exist, we fetch a new sync token for future use.
+        if (syncFile.createNewFile()) {
+            String initialSync = null;
+
+            try {
+                ResultBodyCollection<Event> eventResponse = client.events.get(projectId, "random_sync_token").executeRaw();
+            } catch(InvalidTokenError e) {
+                initialSync = e.sync;
+            }
+
+            writeSyncToken(syncFile, initialSync);
+            System.out.println("First run, saved sync token. Ready for future use.");
+            return;
+        }
+
+        String sync = readSyncToken(syncFile);
+
+        ResultBodyCollection<Event> eventResponse = null;
+
+        try {
+            eventResponse = client.events.get(projectId, sync).executeRaw();
+        } catch(InvalidTokenError e) {
+            writeSyncToken(syncFile, e.sync);
+            System.out.println("Sync token expired or invalid. Saved a new sync token. Ready for future use.");
+            return;
+        }
+        
+        String nextSync = eventResponse.sync;
+        List<Event> events = eventResponse.data;
+
+        if (events.size() == 0) {
+            System.out.println("There are no new events. A new token has been saved.");
+        }
+
+        // Pass through list of events and find stories of completing milestones. Notify slack of completed milestones.
+        for (Event event : events) {
+            if (event.type.equals("story") && event.action.equals("added") && event.resource.resourceType.equals("story")) {
+                Story story = client.stories.findById(event.resource.gid).execute();
```

Comment:
I think we could do this without getting stories on each task if we opt_fields into resource_subtype & use the raw response, which is kind of gross. I'll think about updating the client library to support this.